### DISCUSSION TITLE
Plot DICOM slice and segmentation mask with transparency

### DISCUSSION THREAD

**Discussion by ndahiya3** [pyvista/pyvista#2968] [Q&A]
I am trying to display a dicom slice in plotter with a semi-transparent segmentation mask overlaid in top. I can display the dicom slice by converting it to a uniform grid and adding to plotter with a 'gray' cmap like so:
```
import pyvista as pv
import matplotlib.pyplot as plt
import matplotlib

# Load dicom, get slice and create uniform grid (grid_2d_dcm) with slice data as point data
# Then display grid 
gray_cmap = plt.cm.get_cmap("gray")
actor_dcm = pl.add_mesh(grid_2d_dcm, cmap=gray_cmap)

pl.view_xy()
pl.remove_scalar_bar()
pl.enable_image_style()
pl.show()
```
which gives the following result:
![Dicom_Slice_View](https://user-images.githubusercontent.com/15024623/178175620-3a0c7ec4-6459-4991-b263-970a3a24d3f6.png)

So far so good. Next, instead of the dicom slice I can load the segmentation slice (which is a binary mask), with a new color map which masks empty areas of the binary segmentation mask, and others with 0.5 opacity, like so:
```
import pyvista as pv
import matplotlib.pyplot as plt
import matplotlib

# Load segmentation file, get slice and create uniform grid (grid_2d_seg) with slice data as point data
# Create a new Binary color map
# Then display grid
c_white = matplotlib.colors.colorConverter.to_rgba('white',alpha = 0)
c_red= matplotlib.colors.colorConverter.to_rgba('red',alpha = 0.5)
cmap_wr = matplotlib.colors.ListedColormap([c_white, c_red])

actor_seg = pl.add_mesh(grid_2d_seg, cmap=cmap_wr)

pl.view_xy()
pl.remove_scalar_bar()
pl.enable_image_style()
pl.show()
```
which gives the following expected results:
![Segmentation_View](https://user-images.githubusercontent.com/15024623/178176186-dd4c0d19-171e-477c-9721-f208b9085da7.png)
The empty areas of the segmentation mask don't show up as expected.

But, when I try to overlay this transparent segmentation mask on top of the dicom, the entire image turns red color (as opposed to only masked regions):

```
actor_dcm = pl.add_mesh(grid_2d_dcm, cmap=gray_cmap)
actor_seg = pl.add_mesh(grid_2d_seg, cmap=cmap_wr) # Add seg on top of dicom
```
![Dicom_With_Segmentation](https://user-images.githubusercontent.com/15024623/178176688-78e8a664-21d9-48f8-9df8-759345dbebab.png)

I am really not sure what's going on here. Any ways around this issue? I have tried a bunch of things, but nothing seems to work as expected. Any help on this issue is appreciated.

**Comment by banesullivan** [comment]
Huh. This is very odd. Could you share these data for us to reproduce?

**Reply by ndahiya3** [reply]
Sure. See attached, grayscale `dicom.nii.gz`, binary `segmentation.nii.gz`. My full code is attached below. Thanks!
[dicom.nii.gz](https://github.com/pyvista/pyvista/files/9113411/dicom.nii.gz)
[segmentation.nii.gz](https://github.com/pyvista/pyvista/files/9113414/segmentation.nii.gz)
```
"""
Display Dicom using PyVista

"""

import numpy as np
import SimpleITK as sitk
import pyvista as pv
import matplotlib.pyplot as plt
import matplotlib

# Load Dicom
fname = 'dicom.nii.gz'
dcm_itk = sitk.ReadImage(fname)
dcm_arr = sitk.GetArrayFromImage(dcm_itk)
size = dcm_itk.GetSize()

# Load segmentation
fname = 'segmentation.nii.gz'
seg_itk = sitk.ReadImage(fname)
seg_arr = sitk.GetArrayFromImage(seg_itk)

# Create pseudo 3D uniform grids to hold dicom/segmentation slice
grid_2d_dcm = pv.UniformGrid(dims=(size[0], size[1], 1))
grid_2d_seg = pv.UniformGrid(dims=(size[0], size[1], 1))

# To display 10th slice
arr = np.flipud(dcm_arr[10]).flatten(order='C')
grid_2d_dcm.point_data['ImageFile'] = arr.flatten(order='C')

arr = np.flipud(seg_arr[10]).flatten(order='C')
grid_2d_seg.point_data['ImageFile'] = arr.flatten(order='C')

pl = pv.Plotter()

gray_cmap = plt.cm.get_cmap("gray")

# Binary color map with transparency (seems to works when only segmentation slice is displayed)
c_white = matplotlib.colors.colorConverter.to_rgba('white',alpha = 0)
c_red= matplotlib.colors.colorConverter.to_rgba('red',alpha = 0.5)
cmap_wr = matplotlib.colors.ListedColormap([c_white, c_red])

actor_dcm = pl.add_mesh(grid_2d_dcm, cmap=gray_cmap) # ``cmap`` works correctly while adding only one of the two meshes, not both together
actor_seg = pl.add_mesh(grid_2d_seg, cmap=cmap_wr)

pl.view_xy()
pl.show_axes()
pl.remove_scalar_bar()
pl.enable_image_style()
pl.show()
```
